src/Simplify.elm

```diff
@@ -4355,6 +4369,92 @@ determineIfCollectionIsEmpty moduleName singletonNumberOfArgs lookupTable node =
                 Nothing
 
 
+isComparableValue : Node Expression -> Bool
+isComparableValue (Node _ expression) =
+    case expression of
+        Expression.Integer _ ->
+            True
+
+        Expression.Hex _ ->
+            True
+
+        Expression.Floatable _ ->
+            True
+
+        Expression.Negation expr ->
+            isComparableValue expr
+
+        Expression.Literal _ ->
+            True
+
+        Expression.CharLiteral _ ->
+            True
+
+        Expression.ParenthesizedExpression expr ->
+            isComparableValue expr
+
+        Expression.TupledExpression exprs ->
+            List.all isComparableValue exprs
+
+        Expression.ListExpr exprs ->
+            List.all isComparableValue exprs
+
+        _ ->
+            False
+
+
+simplifyComparableExpression : Node Expression -> List Expression
+simplifyComparableExpression =
+    simplifyComparableExpressionHelper 1
+
+
+simplifyComparableExpressionHelper : Int -> Node Expression -> List Expression
+simplifyComparableExpressionHelper sign (Node _ expression) =
```

Comment:
I think it would be interesting to also support `String`s. Maybe you can combine `simplifyComparableExpression/simplifyComparableExpressionHelper` and `isComparableValue` into one function that folds over expressions and depending on whether we've seen strings or numbers we combine them?

You could "parse don't validate" so that you don't iterate over the list twice. Also, you can use `Normalize.getNumberValue` from this repo to do the same thing you're doing here maybe?

Reply:
Strings and chars are supported, they are in the `_`. I can afford to do that because I know at this point that they are all comparable values. Now I see it would have been clearer to list them explicitly. 

To avoid iterating twice, I think I could re-implement `Maybe.Extra.traverse : (a -> Maybe b) -> List a -> Maybe (List b)`. I'll do that :)